src/db/sqlite.rs

```diff
@@ -1,55 +1,162 @@
     }
 
     fn insert_run(&self, timestamp: &str, tx_count: i64, duration: i64) -> Result<()> {
-        let query = format!(
-            "INSERT INTO runs (timestamp, tx_count, duration) VALUES ({}, {}, {})",
-            timestamp, tx_count, duration
-        );
-        self.conn
-            .execute(&query)
-            .map_err(|_| ContenderError::DbError("failed to insert run", Some(query)))?;
-        Ok(())
+        self.execute(
+            "INSERT INTO runs (timestamp, tx_count, duration) VALUES (?, ?, ?)",
+            params![timestamp, tx_count, duration],
+        )
     }
 
     fn num_runs(&self) -> Result<i64> {
-        let mut stmt = self
-            .conn
-            .prepare("SELECT COUNT(*) FROM runs")
-            .map_err(|_| ContenderError::DbError("failed to prepare query", None))?;
-        stmt.next()
-            .map_err(|_| ContenderError::DbError("failed to execute query", None))?;
-        let count = stmt
-            .read::<i64, _>(0)
-            .map_err(|_| ContenderError::DbError("failed to read result", None))?;
+        let count: i64 =
+            self.query_row("SELECT COUNT(*) FROM runs", params![], |row| row.get(0))?;
         Ok(count)
     }
+
+    fn insert_named_tx(
+        &self,
+        name: String,
+        tx_hash: TxHash,
+        contract_address: Option<Address>,
+    ) -> Result<()> {
+        self.execute(
+            "INSERT INTO named_txs (name, tx_hash, contract_address) VALUES (?, ?, ?)",
+            params![
+                name,
+                tx_hash.encode_hex(),
+                contract_address.map(|a| a.encode_hex())
+            ],
+        )
+    }
+
+    fn get_named_tx(&self, name: &str) -> Result<(TxHash, Option<Address>)> {
+        let pool = self.get_pool()?;
+        println!("name: {}", name);
+        let mut stmt = pool
+            .prepare(
+                "SELECT name, tx_hash, contract_address FROM named_txs WHERE name = ?1 ORDER BY id DESC LIMIT 1",
+            )
+            .map_err(|e| {
+                ContenderError::DbError("failed to prepare statement", Some(e.to_string()))
+            })?;
+        println!("stmt: {:?}", stmt);
+        let row = stmt
+            .query_map(params![name], |row| NamedTxRow::from_row(row))
+            .map_err(|e| ContenderError::DbError("failed to map row", Some(e.to_string())))?;
+        let errr = || ContenderError::DbError("no row", None);
+        let res = row.last().ok_or(errr())?.map_err(|_e| {
+            println!("err {}", _e.to_string());
+            errr()
+        })?;
+        // wtf this feels so wrong
+
+        let tx_hash = TxHash::from_hex(&res.tx_hash)
+            .map_err(|e| ContenderError::DbError("invalid tx hash", Some(e.to_string())))?;
+        let contract_address = res
+            .contract_address
+            .map(|a| Address::from_hex(&a))
+            .transpose()
+            .map_err(|e| ContenderError::DbError("invalid address", Some(e.to_string())))?;
```

Comment:
lmao this is mega ugly. should probably take a nap before attempting to rewrite this